### DISCUSSION TITLE
How to regain functionality from the removed level 1 headline hack?

### DISCUSSION THREAD

**Discussion by SupImDos** [squidfunk/mkdocs-material#5053] [Q&A] [answered]
Hi,

I'm going through the process of updating my documentation from `mkdocs-material` 8.x -> 9.x.

One of the changes I've found is that the header on my index page is no longer detected (as it is below my project logo), and as such the `partials/content.html` is adding an additional `<h1>` tag with the page name.

This can be traced back to commit https://github.com/squidfunk/mkdocs-material/commit/b95dffa5cd7c32a443ab2c46d63252bc828409ec, where the functionality for checking the content for an existing `<h1` tag was removed.

I completely understand the change, and I don't expect this will be reverted. I just wanted to ask if there's a simple way (without overriding `partials/content.html`) that I can regain this functionality and remove the automatic `<h1>`.

For reference, some screenshots below show *before* and *after* this change.
**Before**
![before](https://user-images.githubusercontent.com/62866982/219225323-99ceab51-e169-4e07-a7e9-1cf060b6624d.png)

**After**
![after](https://user-images.githubusercontent.com/62866982/219225347-017189c2-1c14-4cea-b7f7-7c63ee6c2b0a.png)

Thanks!

**Comment by squidfunk** [comment, marked as answer]
Thanks for reporting. The hack was necessary when I started with Material for MkDocs, because MkDocs provided no way of knowing whether an `h1` headline was set or not. However, the use case you describe definitely makes sense: as long as the author provides an `h1` tag (in Markdown or HTML), the headline should not be added to the page's content.

We can revert the relevant section in https://github.com/squidfunk/mkdocs-material/commit/b95dffa5cd7c32a443ab2c46d63252bc828409ec and keep the hack, as it offers more flexibility. However, we can keep the logic in `partials/toc.html`, as the table of contents only includes Markdown headlines anyway. Would you like to craft a PR?

**Reply by SupImDos** [reply]
Thanks @squidfunk, that sounds great.

> Would you like to craft a PR?

I've created a PR for this change here: https://github.com/squidfunk/mkdocs-material/pull/5056. I think I've understood correctly, but let me know if I misunderstood something and you had something else in mind.

**Reply by SupImDos** [reply]
Fixed in #5056, released in `v9.0.13`. This discussion can be marked as solved.
